connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/ErrorFactoryBuilder.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.client
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+private[client] class ErrorFactoryBuilder() {
```

Comment:
TBH, it seems a bit overcomplicated. It's just a map of string <> exception, and you get the exception from the map. If we want something more fancy, I am okay with keeping this. Otherwise, I would just make it simple.

Reply:
I implement the builder class for two reasons:
1. Use addConstructor method to infer the class name of the exceptions when updating the map
2. Use a builder class for extensions. There may be other places that need to register a couple of Exceptions. Having a builder class may such extensions easier.